File: SproutReferenceBot/SproutReferenceBot/Services/BotService.cs
using SproutReferenceBot.Enums;
using SproutReferenceBot.Models;

namespace SproutReferenceBot.Services;
  class BotService
  {
    private Guid BotId;
    private BotStateDTO LastKnownState;
    
    private static readonly int SquareSize = 5;

    private static readonly BotAction[] ActionOrder = new[]
    {
      BotAction.Up,
      BotAction.Right,
      BotAction.Down,
      BotAction.Left,
    };

    private int CurrentAction = 0;
    private int StepsTaken = 0;

    public BotCommand ProcessState(BotStateDTO BotState)
    {
      if (StepsTaken++ >= SquareSize)
      {
        CurrentAction = (CurrentAction + 1) % ActionOrder.Length;
        StepsTaken = 0;
      }

      var ActionToTake = ActionOrder[CurrentAction];
      
      return new BotCommand
      {
        BotId = BotId,
        Action = ActionToTake,
      };
    }

    public void SetBotId(Guid NewBotId)
    {
      BotId = NewBotId;
    }
  }

File: StarterBots/CPPBot/service/BotService.cpp
#include "BotService.h"
#include "../model/BotState.h"

BotService::BotService() {

}

void BotService::SetBotId(std::string botId) {
  this->botId = botId;
}

BotCommand BotService::GetCommand() const {
  return BotCommand{
    .botId = botId,
    .action = nextAction
  };
}

void BotService::ProcessState(const BotState& botState) {
  // TODO: Implement your bot logic here!
  nextAction = BotAction::Up;
}
